### PR TITLE
Add a function to return the default ACL for a directory

### PR DESCRIPTION
**Pull request by aidanhs**

### CODE REVIEW

**Comment by intgr** [comment]
Thanks, will try to take a look at this later today.

File: src/lib.rs [hunk 1]
```diff
@@ -337,6 +348,15 @@ impl PosixACL {
         }
         Ok(())
     }
+
+    /// Returns a raw `acl_t` type that can be used directly in FFI calls to the acl library.
+    ///
+    /// The caller *must not*:
+    /// - `acl_free` or otherwise invalidate the returned `acl_t`
+    /// - use this `acl_t` after the PosixACL has been dropped
+    pub fn as_raw(&self) -> acl_sys::acl_t {
```
Comment (intgr) [inline src/lib.rs]: It seems this method should be marked `unsafe`, what do you think?

Just wondering, what use case do you have for this? Maybe it makes sense to implement your needed feature directly in posix-acl, in addition to this.

File: src/lib.rs [hunk 2]
```diff
@@ -188,14 +188,25 @@ impl PosixACL {
         PosixACL { acl }
     }
 
-    /// Read a file's ACL and return as `PosixACL` object.
+    /// Read a path's ACL and return as `PosixACL` object.
```
Comment (intgr) [inline src/lib.rs:191]: :+1:

File: src/lib.rs [hunk 3]
```diff
@@ -188,14 +188,25 @@ impl PosixACL {
...
     pub fn read_acl(path: &Path) -> Result<PosixACL, SimpleError> {
+        Self::read_acl_flags(path, acl_sys::ACL_TYPE_ACCESS)
+    }
+
+    /// Read a path's default ACL and return as `PosixACL` object.
```
Comment (intgr) [inline src/lib.rs]: Since this doesn't support files, how about "`Read a directory's default ACL ...`"

**Review by intgr — CHANGES_REQUESTED** [review]
Thanks for this. :)

I'd prefer if the two changes are split into two separate pull requests.

PS: Since you probably have more experience with Rust than I do, any feedback you have about posix-acl would be much appreciated.

Meanwhile, I will need to fix my CI config to run tests for pull requests from foreign repositories 🙈

**Comment by intgr** [comment]
Hopefully my CI configuration now works after #19. If you rebase/merge with latest master branch.

**Comment by intgr** [comment]
Hi @aidanhs, are you still interested in finishing these? There's no rush, just wondering.

File: src/lib.rs (hunk 3 above)
Comment (aidanhs) [reply src/lib.rs]: Fixed

File: src/lib.rs (hunk 1 above)
Comment (aidanhs) [reply src/lib.rs]: Moved discussion to other PR.

**Comment by aidanhs** [comment]
I've split out the as_raw work to a separate PR and updated this one.

On feedback - I'm just really happy posix-acl exists so I don't have to create it myself! It hasn't got in my way at all, which I consider pretty important :smile:. The only thing that was a bit odd was the choice of SimpleError as an Error type - it's not an issue for my use-case, but others may prefer something more structured (`io::Error` could be an useful start, as POSIX ACL libraries look like they use the system errno rather than their own type).

**Review by intgr — APPROVED** [review]

**Comment by intgr** [comment]
Merged, thanks!

> The only thing that was a bit odd was the choice of SimpleError as an Error type

That's a good point, this library started out as part of a CLI app [ego](https://github.com/intgr/ego) and it didn't occur to me to change error handling when I split it out.

I'll have to look into error handling patterns in Rust. `io::Error` does look like an option, indeed.

**Comment by intgr** [comment]
I've opened issue #22 to track that progress.

**Comment by intgr** [comment]
@aidanhs FYI I released version 0.4.0 with this feature and a bunch of other stuff.

**Comment by aidanhs** [comment]
@intgr thanks!

**Pull request merged**